src/modules/cpu/kernel/mel_filter_bank.hpp

```diff
@@ -0,0 +1,252 @@
+
+        return mel;
+    }
+
+    Rpp32f mel_to_hz(Rpp32f mel)
+    {
+        Rpp32f hz = 0.0f;
+        if (mel >= min_log_mel)
+            hz = min_log_hz * std::exp(step_log * (mel - min_log_mel));
+        else
+            hz = freq_low + mel * fsp;
+        return hz;
+    }
+    public:
+        ~SlaneyMelScale() {};
+};
+
+RppStatus mel_filter_bank_host_tensor(Rpp32f *srcPtr,
+                                      RpptDescPtr srcDescPtr,
+                                      Rpp32f *dstPtr,
+                                      RpptDescPtr dstDescPtr,
+                                      Rpp32s *srcDimsTensor,
+                                      Rpp32f maxFreqVal,
+                                      Rpp32f minFreqVal,
+                                      RpptMelScaleFormula melFormula,
+                                      Rpp32s numFilter,
+                                      Rpp32f sampleRate,
+                                      bool normalize,
+                                      rpp::Handle& handle)
+{
+    BaseMelScale *melScalePtr;
+    switch(melFormula)
+    {
+        case RpptMelScaleFormula::HTK:
+            melScalePtr = new HtkMelScale;
+            break;
+        case RpptMelScaleFormula::SLANEY:
+        default:
+            melScalePtr = new SlaneyMelScale();
+            break;
+    }
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        Rpp32f *dstPtrTemp = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        Rpp32s samples = srcDimsTensor[batchCount * 2];
+        Rpp32s channels = srcDimsTensor[batchCount * 2 + 1];
+
+        // Extract nfft, number of Frames, numBins
+        Rpp32s nfft = (samples - 1) * 2;
+        Rpp32s numBins = nfft / 2 + 1;
+        Rpp32s numFrames = channels;
+
+        Rpp32f maxFreq = maxFreqVal;
+        Rpp32f minFreq = minFreqVal;
+        maxFreq = sampleRate / 2;
+
+        // Convert lower, higher frequencies to mel scale
+        Rpp64f melLow = melScalePtr->hz_to_mel(minFreq);
+        Rpp64f melHigh = melScalePtr->hz_to_mel(maxFreq);
+        Rpp64f melStep = (melHigh - melLow) / (numFilter + 1);
+        Rpp64f hzStep = static_cast<Rpp64f>(sampleRate) / nfft;
+        Rpp64f invHzStep = 1.0 / hzStep;
+
+        Rpp32s fftBinStart = std::ceil(minFreq * invHzStep);
+        Rpp32s fftBinEnd = std::ceil(maxFreq * invHzStep);
+        fftBinEnd = std::min(fftBinEnd, numBins);
+
+        Rpp32f *weightsDown = static_cast<Rpp32f *>(calloc(numBins, sizeof(Rpp32f)));
+        Rpp32f *normFactors = static_cast<Rpp32f *>(malloc(numFilter * sizeof(Rpp32f)));
+        std::fill(normFactors, normFactors + numFilter, 1.f);
+
+        Rpp32s *intervals = static_cast<Rpp32s *>(malloc(numBins * sizeof(Rpp32s)));
```

Comment:
Pls use scratch if there are many

Reply:
Can you please explain this further
what does scratch mean here?

Reply:
@snehaa8 scratch buffer is any memory allocated when our handle is set. You should be able to access https://github.com/r-abishek/rpp/blob/master/src/modules/handlehost.cpp#L50
Please check with @sampath1117 

Evaluate what max size we might need and set bounds so we don't go beyond.

Reply:
Batchsize can go upto max of 128 and maxSrcLength can go upto 7l, so combined normFactors, weightsDown and internvals into one tempFloatmem cpu handle memory.
tempFloatmem can handle upto 99532800 * batchSize.